src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -320,11 +321,19 @@ private void MethodChecks(MethodDeclarationSyntax syntax, Binder withTypeParamsB
 
                     if ((object)overriddenMethod != null)
                     {
-                        CustomModifierUtils.CopyMethodCustomModifiers(overriddenMethod, this, out _lazyReturnType, 
-                                                                      out _lazyCustomModifiers, 
+                        CustomModifierUtils.CopyMethodCustomModifiers(overriddenMethod, this, out _lazyReturnType,
+                                                                      out _lazyCustomModifiers,
                                                                       out _lazyParameters, alsoCopyParamsModifier: true);
                     }
                 }
+                else if (_refKind == RefKind.RefReadOnly)
```

Comment:
>else if (_refKind == RefKind.RefReadOnly) [](start = 16, length = 41)

I think this code should be moved before the ```if (syntax.ExplicitInterfaceSpecifier == null)``` above and combined with code ```_lazyCustomModifiers = CustomModifiersTuple.Empty;``` on line 298. Otherwise, we end up without the modifier for error cases when an overriden or implemented method is not found. Please add tests for scenarios like that.